packages/main/src/SliderBase.js

```diff
@@ -0,0 +1,492 @@
+
+		// If the step value is not a round number get its precision
+		const stepPrecision = SliderBase._getDecimalPrecisionOfNumber(stepSize);
+		return value.toFixed(stepPrecision);
+	}
+
+	/**
+	 * Gets pageX value from event on user interaction with the Slider
+	 *
+	 * @private
+	 */
+	static _getPageXValueFromEvent(event) {
+		if (event.targetTouches && event.targetTouches.length > 0) {
+			return event.targetTouches[0].pageX;
+		}
+		return event.pageX;
+	}
+
+	/**
+	 * Computes the new value (in %) from the pageX position of the cursor.
+	 * Returns the value with rounded to a precision of at most 2 digits after decimal point.
+	 *
+	 * @private
+	 */
+	static _computedValueFromPageX(pageX, min, max, boundingClientRect) {
+		// Determine pageX position relative to the Slider DOM
+		const xPositionRelative = pageX - boundingClientRect.left;
+		// Calculate the percentage complete (the "progress")
+		const percentageComplete = xPositionRelative / boundingClientRect.width;
+		// Fit (map) the complete percentage between the min/max value range
+		return min + percentageComplete * (max - min);
+	}
+
+	/**
+	 * Calculates the precision (decimal places) of a number, returns 0 if integer
+	 * Handles scientific notation cases.
+	 * @private
+	 */
+	static _getDecimalPrecisionOfNumber(value) {
+		if (Number.isInteger(value)) {
+			return 0;
+		}
+		const match = (String(value)).match(/(?:\.(\d+))?(?:[eE]([+-]?\d+))?$/);
+		return Math.max(0, (match[1] ? match[1].length : 0) - (match[2] ? Number(match[2]) : 0));
+	}
+
+	/**
+	 * Update initial Slider UI representation and normalize internal state
+	 * Normalize Range Slider values according to min/max properties
+	 * Normalize the step value and draw tickmarks/labels if specified
+	 * 
+	 * Returns <code>true</code> if UI has to be updated further to sync 
+	 * with the internal state and <code>undefined</code> otherwise
+	 * @private
+	 */
+	_syncUIAndState() {
+		// In this case the value prop is changed programatically (not by user interaction)
+		// and it won't be "stepified" (rounded to the nearest step)
+		if (this.step !== this._prevStepValue) {
+			this._setStep(this.step);
+			this._prevStepValue = this.step;
+			return;
+		}
+
+		if (this.min !== this._prevMin || this.max !== this._prevMax) {
+			this._drawDefaultTickmarks(this.step, this.max, this.min);
+			this._prevMin = this.min;
+			this._prevMax = this.max;
+			return true;
+		}
+	}
+
+	/**
+	 * Calculates and draws the tickmarks with a CSS gradient style
+	 *
+	 * @private
+	 */
+	_drawDefaultTickmarks(step, max, min) {
+		if (!this.tickmarks || !this.step) {
+			return;
+		}
+
+		// Let the CSS do calculations for precise tickmarks distribution
+		const stepStr = String(step);
+		const maxStr = String(max);
+		const minStr = String(min);
+		const tickmarkWidth = "1px";
+
+		this._tickmarksAmount = `${maxStr - minStr} / ${stepStr}`;
```

Comment:
Why do these need to get converted into Strings?
${maxStr - minStr} would be calculated

Reply:
To let the CSS do the calculation, for "better rounding/subpixel behavior". Seems like this way produces more precise (the best possible) positioning of the tickmarks.